UnitTestFrameworkPkg/Library/UnitTestDebugAssertLib/UnitTestDebugAssertLibHost.cpp

```diff
@@ -24,6 +24,27 @@ extern "C" {
   ///
   BASE_LIBRARY_JUMP_BUFFER  *gUnitTestExpectAssertFailureJumpBuffer = NULL;
 
+  /**
+    LongJump wrapper for host-based unit test environments that is declared
+    NORETURN to avoid false positives from address sanitizer.
+
+    @param  JumpBuffer  A pointer to CPU context buffer.
+    @param  Value       The value to return when the SetJump() context is
+                        restored and must be non-zero.
+  **/
+  static
+  VOID
+  NORETURN
+  EFIAPI
+  HostLongJump (
+    IN      BASE_LIBRARY_JUMP_BUFFER  *JumpBuffer,
+    IN      UINTN                     Value
+    )
+  {
+    LongJump (JumpBuffer, Value);
+    UNREACHABLE ();

```

Comment:
Why is this needed? Is LongJump not marked as noreturn?

Reply:
Correct.  LongJump() in MdePkg is not marked as NORETURN.  I looked down that path and it then requires all the APIs that call LongJump() to also be declared as NORETURN and would be a very large change to the entire edk2 sources.